api/types/fix.go

```diff
@@ -9,14 +9,15 @@ import (
 	"sigs.k8s.io/yaml"
 )
 
-// FixKustomizationPreUnmarshalling modies the raw data
+// FixKustomizationPreUnmarshalling modifies the raw data
 // before marshalling - e.g. changes old field names to
 // new field names.
 func FixKustomizationPreUnmarshalling(data []byte) ([]byte, error) {
-	deprecateFieldsMap := map[string]string{
-		"imageTags:": "images:",
+	deprecatedFieldsMap := map[string]string{
+		"imageTags:":      "images:",
+		"patchesJson6902": "patches",
 	}
-	for oldname, newname := range deprecateFieldsMap {
+	for oldname, newname := range deprecatedFieldsMap {
 		pattern := regexp.MustCompile(oldname)
 		data = pattern.ReplaceAll(data, []byte(newname))
 	}
```

Comment:
Some tests are failing.  

Maybe because this PR doesn't convert  instances of `PatchJson6902`
https://github.com/kubernetes-sigs/kustomize/blob/master/api/types/patchjson6902.go

to instances of `Patch`
https://github.com/kubernetes-sigs/kustomize/blob/master/api/types/patch.go

These structs are almost the same - both have a `Path` and `Patch` field,
but the `Target` fields are different.  PatchJson6902 also allows one to 
specify label and annotation selectors.

Reply:
Looking into test failures 
```FAIL: TestMakeCustomizedResMap (0.03s)
    kusttarget_test.go:255: unexpected Resources error add operation does not apply: doc is missing path: "/spec/replica": missing value
```
and
```FAIL: TestDisableNameSuffixHash (0.01s)
    harness.go:102: add operation does not apply: doc is missing path: "/spec/replica": missing value
``` 
now.
I believe it's `Patch`'s `Target` field that allows the additional selectors based on 
https://github.com/kubernetes-sigs/kustomize/blob/master/api/types/selector.go